### PR TITLE
Features/keycloak integration

### PR DESCRIPTION
**Pull request by JerryPan44**
Initial review keycloak

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -33,7 +33,8 @@ services:
       - db
 
   portalbackend:
-    image: hbpmip/portal-backend:5.0.2
+    #image: hbpmip/portal-backend:5.0.5
+    image: jerrypan44/portal-backend:latest
```
Comment (nicedexter) [inline docker-compose.yml]: we need a published image with a release tag under hbpmip/portal-backend:5.1.0

File: docker-compose.yml [hunk 2]
```diff
@@ -44,29 +45,53 @@ services:
...
+      #USER_INFO_URI: "http://88.197.53.10:8095/auth/realms/Demo/protocol/openid-connect/userinfo"
+      #TOKEN_URI: "http://88.197.53.10:8095/auth/realms/Demo/protocol/openid-connect/token"
+      #LOGOUT_URI: "http://88.197.53.10:8095/auth/realms/Demo/protocol/openid-connect/logout"
+      
+      #remote keycloak
+      #AUTH_URI: "https://iam.humanbrainproject.eu/auth/realms/MIPTEST/protocol/openid-connect/auth"
+      #USER_INFO_URI: "https://iam.humanbrainproject.eu/auth/realms/MIPTEST/protocol/openid-connect/userinfo"
+      #TOKEN_URI: "https://iam.humanbrainproject.eu/auth/realms/MIPTEST/protocol/openid-connect/token"
+      
+      KEYCLOAK_URL: "88.197.53.106"
```
Comment (nicedexter) [inline docker-compose.yml]: cleanup needed

File: docker-compose.yml [hunk 3]
```diff
@@ -77,7 +102,7 @@ services:
       - portalbackend
 
   frontend:
-    image: hbpmip/portal-frontend:5.1.2
+    image: hbpmip/portal-frontend:5.1.11-58-g15c70130
```
Comment (nicedexter) [inline docker-compose.yml]: hbpmip/portal-frontend:5.1.3

File: docker-compose.yml [hunk 4]
```diff
@@ -33,7 +35,8 @@ services:
       - db
 
   portalbackend:
-    image: hbpmip/portal-backend:5.0.2
+    #image: hbpmip/portal-backend:5.0.4
+    image: jerrypan44/portal-backend:latest
```
[the remark this answers is not in the document]
Comment (ThanKarab) [reply docker-compose.yml]: Do you mean 5.0.1? Does the 5.1.0 exist for the portal-backend? 
https://hub.docker.com/repository/registry-1.docker.io/hbpmip/portal-backend/tags?page=1
I can't see it here.

File: docker-compose.yml (hunk 1 above)
Comment (ThanKarab) [reply docker-compose.yml]: Shouldn't we do this after we merge?

We usually merge the code if it looks good and then publish the images. What do you think?

Comment (ThanKarab) [reply docker-compose.yml]: Do you just want to have it like that to be ready for the post-merge images?

**Comment by nicedexter** [comment]
A new version has to be created with the keycloack implementation in order to be deployed.

Because it’s a new functionality and it’s backward compatible, - according to https://semver.org/ -, it seems ok to name it 5.1

**Comment by nicedexter** [comment]
Yes exactly. We need the new backend image published to update this repo

**Pull request merged**